lib/puppet/functions/letsencrypt_lookup.rb

```diff
@@ -0,0 +1,8 @@
+Puppet::Functions.create_function(:letsencrypt_lookup) do
```

Comment:
can you please namespace the function like `:'letsencrypt::lookup'`? The file needs to be moved to a `letsencrypt` subdirectory for this. Topscope functions are deprecated and we would like to not introduce more of them

Reply:
I mirrored what they did in puppetlabs/apache https://github.com/puppetlabs/puppetlabs-apache/blob/master/lib/puppet/functions/apache/apache_pw_hash.rb